app/views/rankings/index.html.erb

```diff
@@ -1,6 +1,10 @@
 <h1><%= ranking_title(params[:time_period]) %></h1>
 <p><%= render partial: 'shared/ranking_links' %></p>
-<div class="leaderboard_list">
-  <%= render :partial => 'ranking', :locals => {:leaderboard => @leaderboard} %>
-</div>
-<p><strong>Note:</strong> leaderboard updates every <%= distance_of_time_in_words(@cache_expiration) %>.</p>
\ No newline at end of file
+
+<% cache [:user_ranking_partial, params[:time_period]] do %>
+  <%= render :partial => 'ranking', :locals => { :leaderboard => @leaderboard } %>
+<% end %>
```

Comment:
This doesn't work well if we want to show a different name (i.e. "you") for authenticated users.